### PR TITLE
Modify RewriteRule to accept callable rhs

### PR DESCRIPTION
**Pull request by jcrist**
`RewriteRule` now accepts callable `rhs` arguments. If the `rhs` is callable,
it's called with a `dict` of matches. If the `rhs` is a task, the
matches are substituted in to the `rhs`.

### CODE REVIEW

**Comment by mrocklin** [comment]
Is there a significant performance change?

File: dask/rewrite.py [hunk 1]
```diff
@@ -133,8 +133,11 @@ class RewriteRule(object):
     ----------
     lhs : task
         The left-hand-side of the rewrite rule.
-    rhs : task
-        The right-hand-side of the rewrite rule.
+    rhs : task or function
+        The right-hand-side of the rewrite rule. If it's a task, variables in
+        `rhs` will be replaced by terms in the subject that match the variables
+        in `lhs`. If it's a function, the function will be called with a dict
+        of such matches.
```
Comment (mrocklin) [inline dask/rewrite.py]: Is there an easy example one could add to the RewriteRule docstring showing how this works?

Comment (jcrist) [inline dask/rewrite.py]: Yeah, I can pull in the one from the tests.

**Comment by jcrist** [comment]
There's no performance change, the branch is determined on rule creation - during rewriting everything is the same is before.

**Comment by mrocklin** [comment]
This is working like a champ at #103 by the way.  There are some things to iron out with my attempt at merging slices but otherwise this is looking good.

It feels satisfying to find pragmatic use cases of theoretically powerful tools.

**Pull request merged**